crates/client/src/client.rs

```diff
@@ -354,6 +355,14 @@ impl FuelClient {
         self.query(query).await.map(|r| r.latest_gas_price.into())
     }
 
+    pub async fn estimate_gas_price(
+        &self,
+        block_horizon: u32,
```

Comment:
Would it make more sense to accept `block_horizon` as an instance of `BlockHorizon` here, rather than converting in the function itself? Or maybe some `T` that can `.into()` into a `BlockHorizon`?